backend/typescript/services/implementations/reviewService.ts

```diff
@@ -399,6 +480,266 @@ class ReviewService implements IReviewService {
 
     return result;
   }
+
+  async updateReviews(id: string, entity: ReviewRequestDTO): Promise<void> {
+    try {
+      await this.db.transaction(async (t) => {
+        const reviewToUpdate = await PgReview.findByPk(id, {
+          transaction: t,
+          include: [{ all: true, nested: true }],
+        });
+
+        if (!reviewToUpdate) {
+          throw new Error(`Review id ${id} not found`);
+        }
+
+        // delete tags
+        const newTagIds = entity.tags
+          .filter((tag: TagRequest) => typeof tag.id === "number")
+          .map((tag: TagRequest) => tag.id);
+        const allReviewTags = await PgReviewTag.findAll({
+          where: { review_id: id },
+          transaction: t,
+        });
+        const droppedTags = allReviewTags
+          .map((reviewTag: PgReviewTag) => reviewTag.tag_id)
+          .filter((oldId: number) => !newTagIds.includes(oldId));
+        await Promise.all(
+          droppedTags.map(async (tagId: number) => {
+            const dbTag = await PgTag.findByPk(tagId, { transaction: t });
+            if (!dbTag) {
+              throw new Error(`Tag id ${dbTag} not found`);
+            }
+            reviewToUpdate.$remove("tags", dbTag, { transaction: t });
+          }),
+        );
```

Comment:
can we move each one of these blocks into separate helper functions?

Reply:
We could but are we going to benefit much from it? @susheelk  what do you think?

Reply:
I think ideally we should for the sake of cleanliness, but we need this PR merged ASAP. Only do so if it takes you <1hr and doesn't require re-testing.